src/base/bittorrent/torrenthandle.cpp

```diff
@@ -1688,7 +1688,18 @@ void TorrentHandle::handleTrackerErrorAlert(const lt::tracker_error_alert *p)
 
     m_trackerInfos[trackerUrl].lastMessage = message;
 
-    m_session->handleTorrentTrackerError(this, trackerUrl);
+    // Starting with libtorrent 1.2.x each tracker has multiple local endpoints from which
+    // an announce is attempted. Some endpoints might succeed while others might fail.
+    // Emit the signal only if all endpoints have failed. TrackerEntry::isWorking() returns
+    // true if at least one endpoint works.
+    for (const TrackerEntry &entry : asConst(trackers())) {
+        if (trackerUrl == entry.url()) {
+            if (!(entry.nativeEntry().verified && entry.isWorking()))
```

Comment:
Perhaps `nativeEntry()` shouldn't be called outside of `TrackerEntry`? And should add an getter method in `TrackerEntry` for the `verified` field.
I'm not sure so I'm leaving a note here.

Reply:
>Perhaps nativeEntry() shouldn't be called outside of TrackerEntry? And should add a getter method in TrackerEntry for the verified field.

Probably. And I think @glassez would want it too, right?

Reply:
>Perhaps nativeEntry() shouldn't be called outside of TrackerEntry?

Yes. It's for passing into libtorrent functions.

Reply:
>And should add a getter method in TrackerEntry for the verified field.

You can just test for `TrackerEntry::Working` status instead.

Reply:
>You can just test for TrackerEntry::Working status instead.

Sure, but `TrackerEntry::status()` does a lot more tests internally... Is that ok?

Reply:
Well, then just implement isVerified()`.

BTW, it seems wrong that we have different "Working" meanings in TrackerEntry interface.